doc/getting-started.md

```diff
@@ -0,0 +1,344 @@
+
+1. [Run a dogecoin node](#run-a-dogecoin-node)
+2. [Introduction to JSON-RPC API](#introduction-to-json-rpc-api)
+    * [Create a wallet](#create-a-wallet)
+    * [Send a transaction](#send-a-transaction)
+    * [Inspect blocks and transactions](#inspect-blocks-and-transactions)
+3. [Node configuration](#node-configuration)
+    * [Mainnet, testnet and regnet](#mainnet-testnet-and-regnet)
+    * [RPC credentials](#rpc-credentials)
+    * [Ports](#ports)
+    * [Memory](#memory)
+    * [Mode debug](#mode-debug)
+
+## Run a dogecoin node
+
+To launch your node, you can choose `dogecoind` to run an headless server or `dogecoin-qt` for a server with a GUI.
+
+In a terminal, run :
+```bash
+# Headless node
+$ dogecoind -printtoconsole
+
+# GUI
+$ dogecoin-qt
+```
+
+*Use `--help` to see all available options for each executables.*
+
+Your node should now be running, starting a *syncing process* by downloading the entire blockchain from other nodes. This operation will take hours, days to complete... but your now part of the dogecoin network !
+
+> **Warning :** The rest of the guide will assume the use of an headless node. The RPC server is not exposed with `dogecoin-qt` until you activate the `-server` option.
+
+## Introduction to JSON-RPC API
+
+Dogecoin Core expose a JSON-RPC interface. With it, you can request information about blocks and transactions, send transactions to the networks, manage your wallet and so on...
+
+To interact with the interface, it relies on HTTP requests. Using urls like `http://user:password@127.0.0.1:22555/` and sending commands through json format is feasible.
+
+Some tools like libraries can be built on top of it. `dogecoin-cli` is the default shell client provided with Dogecoin Core to use this API.
+
+The RPC API documentation is available from [bitcoin](https://developer.bitcoin.org/reference/rpc/).  
+To have an overview of all commands, see also :
+
+```bash
+# List all commands
+$ dogecoin-cli help
+
+# Get help for a specific command
+$ dogecoin-cli help COMMAND
+```
+> **Tips** : Abuse of the help menus is allowed ! They can answer to multiple questions.
+
+### Create a wallet
+
+To receive or recover coins, you will need a wallet address and a private key. The *wallet address* is a shareable information to receive dogecoins from all over the world, while the *private key* is a sensitive information who give the ownership on coins associated to a public address.
+
+By default, you already have a wallet address and a private key available.
+
+List it using `getaddressesbyaccount` :
+
+```bash
+$ dogecoin-cli getaddressesbyaccount ""
+[
+  "DBZc6qeBuo7g8La7L3Lhaf39MsJSgbzpMX"
+]
+```
+
+Use `getnewaddress` to generate a new wallet address :
+```bash
+$ dogecoin-cli getnewaddress
+DC13A8qUgAu5rTWweb44TwXZNQbxDYhyo2
+```
+
+And `dumpprivkey` to find the corresponding private key.
+```bash
+$ dogecoin-cli dumprivkey DC13A8qUgAu5rTWweb44TwXZNQbxDYhyo2
+QSqPGRwxeqv8KNEU85jw128g7rt5ognjqaSvjmeouuFAjJ8Bq3ZR
+```
```

Comment:
It depends on who is getting started. When I was doing it I've more developers in mind rather than common users. From this point of view, generating private keys, list them, import and dump them are basics commands to play with Dogecoin Core.

Reply:
For developers, I would write a completely different guide though and have different guides for different development goals. For example, someone handling public coin should never use the wallet because it forces a solution to be custodial. Someone developing integration with their storefront maybe could use the wallet, but I'd still be extremely cautious with recommending that because you really don't want your keys exposed like that. A developer should learn how to use extended public keys so that they can segregate secrets from address generation - and maybe even keep their keys safely in a HW wallet.

Let's clarify this first because it impacts the entire document: devs is not the same audience as the "technical users" you quoted in your summary. You want this to be pedagogical and beginner friendly? Then it should teach the right things, safe things, and not dangerous things.

Reply:
`dumpprivkey` is a basic command, it's definitively needed to control your private key. It creates security risks for sure, but I feel it's more a security by ignorance and I really dislike it. "Not your keys, not your coins" => No access to your keys, no control of your coins.

People have to educate themselves about private/public key. This command is also coming from [Mastering Bitcoin](https://github.com/bitcoinbook/bitcoinbook/blob/develop/ch05.asciidoc).

Reply:
Also this get is definitively not targeting some kind of basic hodler. I suppose main Dogecoin usage are :
1. people Using exchanges
2. People using non-custodial
3. People running a node with their wallet, using GUI
4. People running a node with their wallet, using CLI.

Like it's really specific type of user who will come here, and when I think about Dogecoin core user it's more someone with programming skills already, people who will read it are filtered.

Reply:
So from my perspective, this guide basically covers advanced users under 3 and 4 (3 because any command with `dogecoin-cli` can be done on the RPC console in QT) Maybe someone that wants to automate some personal coin sends... such as *whenever Elon tweets, send some Dogecoin to my fav exchange so I can sell at peak and buy back low later*.

> People have to educate themselves about private/public key. This command is also used in Mastering Bitcoin.

That section also mentions that there are HD wallets that have key derivation. That's what we have (basically right now we have what they call Type-1 HD wallets). For these, being careless with private keys can really hurt because of the way BIP32 derivation works. One key and some signatures can in theory expose the entire chain of keys. So this makes it much riskier to expose private keys like that.

> "Not your keys, not your coins" => No access to your keys, no control of your coins.

Owning your keys is not the same as exposing your keys. wallet.dat is a documented bdb container that can be read also outside of Dogecoin Core (in case you were worried about vendor lock-in on an open-source SW) I understand the educational value of what you're trying to achieve, but if you want to do this, then this should not be part of getting started, but part of a "how to secure your coin - let's try this on testnet" type of guide. `dumpprivkey` is really the worst possible way to securely manage your keys, because it removes all security and just outputs a secret in plaintext.

Reply:
Is there any other command to export your private key securely with `dogecoin-cli` ? I don't find any, and I'm wondering if encryption is a layer build on top of this command.

Reply:
`backupwallet` doesn't decrypt keys.

Reply:
`dumprivkey` replaced by `backupwallet`.